mcs/class/corlib/System.IO/MonoIO.cs

```diff
@@ -169,32 +169,80 @@ public static Exception GetException (MonoIOError error)
 		// directory methods
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		public extern static bool CreateDirectory (string path, out MonoIOError error);
+		public extern static bool CreateDirectory (IntPtr path, out MonoIOError error);
+
+		public unsafe static bool CreateDirectory (string path, out MonoIOError error)
```

Comment:
Does this really need to be "public unsafe"? Why not just wrap the `fixed` block in an `unsafe` block?

Reply:
Indeed, please wrap the `fixed` inside a `unsafe` block instead of marking the whole method `unsafe`.

Reply:
AFAIK, you yourself have to be an unsafe (function|block) to call an unsafe function.  An unsafe block encapsulates the unsafety.

Reply:
Ah, that’s what I’d originally thought `unsafe` was for, but I grew too accustomed to working in corlib.

Reply:
> AFAIK, you yourself have to be an unsafe (function|block) to call an unsafe function.

This is only true if the method has an unsafe return value or parameter(s), i.e. pointers. `unsafe` doesn't even exist on the IL level. If you declare something like:

```csharp
public unsafe void Foo (int x) { ... }
```

You can totally call that in non-`unsafe` context.